### ISSUE TITLE
📎 Implement `useImplicitBoolean`

### ISSUE THREAD

**Issue by dyc3** [biomejs/biome#5095]
### Description

Add `useImplicitBoolean`, a rule that enforces using implicit booleans in JSX over explicit boolean props. 

Valid
```jsx
<Foo bar />
<Foo bar={false} />
```

Invalid
```
<Foo bar={true} />
```

Requested here: https://github.com/biomejs/biome/discussions/3#discussioncomment-12131883

Source: https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-boolean-value.md

Note that we already have [`noImplicitBoolean`](https://biomejs.dev/linter/rules/no-implicit-boolean/), the opposite of this rule. Would it be worth it to combine them into one rule? The best time to do so would be before 2.0.

**Comment by arendjr** [comment]
`useConsistentBooleanProps` with values `implicit` and `explicit`? I don’t really like that `noExplicitBoolean` has no reference in its name that it’s for use with JSX; it sounds like something TypeScript-specific.

**Comment by ematipico** [comment]
I removed the rule from v2. This can be shipped whenever we can, and it shouldn't stop us from releasing v2

**Comment by izakfilmalter** [comment]
Bit confused here, is there a rule that marks
```tsx
<Foo bar={true} />
```
as invalid?

**Comment by siketyan** [comment]
@izakfilmalter That is what this issue will achieve.

**Comment by vladimir-ivanov** [comment]
@arendjr @siketyan @dyc3 @ematipico 
can we do a rule like:

```json
"useConsistentBooleanProps": {
  "mode": "implicit"   // enforce <Foo bar />
}
// or:
"useConsistentBooleanProps": {
  "mode": "explicit"   // enforce <Foo bar={true} />
}
```

Or do we want React specific (tsx, jsx only).

And what happens with the existing one if we introduce a different rule please? mark as deprecated?

I am a fan of implicit mode for react personally and happy to implement it once there is an agreement on the approach.

Thank you

**Comment by ematipico** [comment]
I think it's a good plan, and we already deprecated rules in favour of others.

One thing that this new rule needs to do is to ignore cases like

```jsx
let disabled = false;
<Foo disabled={disabled}>
```
